### ISSUE TITLE
dynatrace-sli-service: Ensure compatibility with new metrics v2 api (/api/v2/metrics/query)

### ISSUE THREAD

**Issue by johannes-b** [keptn/keptn#1282]
The sli service queries `/api/v2/metrics/series` when it should start talking to `/api/v2/metrics/query`

See query: https://github.com/keptn-contrib/dynatrace-sli-service/blob/d76e0ebff648f53a18ef97da1cd4a0f0e9632dc5/pkg/lib/dynatrace/dynatrace.go#L143

Execute, for example, `curl -X GET "/api/v2/metrics/series/builtin:service.requestCount.total:merge(0):count?scope=tag(hotday-tag-rule:hotday-tag-rule:carts-carts-test)"`

Then you get a warning: _Any usage of /metrics/series should be updated to use /metrics/query_ 

see: {"totalCount":0,"nextPageKey":null,"metrics":{"builtin:service.requestCount.total:merge(0):count":{"series":[]}},"warnings":["/metrics/series is outdated and has been superseded by /metrics/query. Any usage of /metrics/series should be updated to use /metrics/query. The payload has been modernized and `scope` has been replaced by the new `entitySelector`, with a slightly different syntax that requires a mandatory `type` predicate. Note that the new end point no longer supports metrics with different primary entity types being queried together. See the [Swagger Documentation](/rest-api-doc/?urls.primaryName=Environment%20API%20v2#/Metrics/query_1) for more details on the modern endpoint."]}

**Comment by christian-kreuzberger-dtx** [comment]
After investigation together with @bacherfl we found out that

1. The endpoint can no longer be queried using `/api/v1/metrics/series/$METRICNAME?$PARAMS` but using `/api/v1/metrics/query/?metricSelector=$METRICNAME&$PARAMS`
2. The result used to be
```json
{
    "totalCount": 4,
    "nextPageKey": null,
    "metrics": {
        "builtin:service.response.time:percentile(50):merge(0)": {
            "values": [
                {
                    "dimensions": [],
                    "timestamp": 1579090020000,
                    "value": 62036.60017905623
                }
            ]
        }
    }
}
```
but is now structured slightly different
```json
{
    "totalCount": 8,
    "nextPageKey": null,
    "result": [
        {
            "metricId": "builtin:service.response.time:percentile(50):merge(0)",
            "data": [
                {
                    "dimensions": [],
                    "timestamps": [
                        1579097520000
                    ],
                    "values": [
                        65005.48481639812
                    ]
                }
            ]
        }
    ]
}
```
3. The param `scope`  has been renamed to `entitySelector` and it must contain a selector for the entity, e.g., `type(SERVICE)`

In summary, the following steps are required

- [ ] Rewrite the code for the HTTP request in dynatrace-sli-service  (and handle the different response)
- [ ] Rewrite the default SLI configuration for dynatrace-sli-service to the new format
- [ ] Adapt documentation for the SLI configuration for dynatrace-sli-service
- [ ] Adapt example SLI configuration in keptn/examples repo

**Comment by christian-kreuzberger-dtx** [comment]
The old API endpoint will be available for some time, the new endpoint is not yet available. Therefore I've scheduled this issue for release 0.6.1.

**Comment by christian-kreuzberger-dtx** [comment]
@namratachaudhary has sent us this PR: https://github.com/keptn-contrib/dynatrace-sli-service/pull/24

we will merge this once we have finished releasing keptn 0.6.0

**Comment by christian-kreuzberger-dtx** [comment]
I had to slightly adapt the implementation to support both, the old format of `$timeseriesIdentifier?scope=...&...` and the new format with `metricsIdentifier=$timeseriesIdentifier&entitySelector=...` at the same time - see PR https://github.com/keptn-contrib/dynatrace-sli-service/pull/27 .

A preview is available with the following docker image: `keptn/dynatrace-sli-service:feature.1282.20200128.1319`


Still open:

- [ ] Update docs
- [ ] Write integration tests to verify that both formats are still working

**Comment by namratachaudhary** [comment]
Where should the docs be updated? I could take up those two tasks - I'm just not sure where they are supposed to be done.

**Comment by christian-kreuzberger-dtx** [comment]
Hi, 

this is my current PR (that still needs to be reviewed and merged): https://github.com/keptn-contrib/dynatrace-sli-service/pull/27

The following still needs to be updated within that PR (or with a second PR)

**code**:
* [dynatrace/dynatrace.go](https://github.com/keptn-contrib/dynatrace-sli-service/blob/f6745d124ca7d0c1fe3272d753f71ae205e692cd/pkg/lib/dynatrace/dynatrace.go#L308-L330) Change the default metrics to the new format with `metricSelector=builtin:service.requestCount.total:merge(0):count&...`

**docs**:
* [README.md](https://github.com/keptn-contrib/dynatrace-sli-service/blob/master/README.md) update the section with *The default SLI queries for this service are defined as follows:* with the metrics from above
* [docs/DynatraceIntegration.md](https://github.com/keptn-contrib/dynatrace-sli-service/blob/feature/1282/new_metrics_api_compat/docs/DynatraceIntegration.md) - I think this doc also needs some updates for the new api

**Comment by namratachaudhary** [comment]
Sorry about the delay - 
Have updated the Readme and integration document - alongwith static metric definitions in the following PR
https://github.com/keptn-contrib/dynatrace-sli-service/pull/28

If I have missed something, please let me know!
:bowing_woman:
